nuxeo-ai-model/src/main/java/org/nuxeo/ai/adapters/Adapter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (C) Copyright 2006-2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * Contributors:
+ *     anechaev
+ */
+package org.nuxeo.ai.adapters;
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+
+public interface Adapter {
```

Comment:
The name and concept of an `Adapter` is already used in the Nuxeo platform.  I don't understand why you created an interface with that name.

Reply:
it's a default interface that allows to have consistency within the project. It requires derivatives to be able to give back document or other common props across the system.

Reply:
Perhaps `AIAdapter` then?

Reply:
not really. It has nothing to do with AI more like a common interface that should be a common one within platform IMO

Reply:
That's my point.  Its already a concept in the platform but you are adding it in the AI module

Reply:
renamed to `AIAdapter`